### PR TITLE
Remove discussion of opacity of `Integers`

### PR DESCRIPTION
**Pull request by SiriusStarr**
As `Integer`s are no longer opaque, the design consideration documentation required updating.

### CODE REVIEW

File: docs/discussions/Design-choices.md [hunk 1]
```diff
@@ -126,18 +126,18 @@ type).
...
-`Natural`s hold a privileged status within the language because functions based
-on `Natural`s tend be "corner-case-free", meaning that they gracefully handle
-all possible inputs.
-
-`Integer` and `Double`s are opaque values as far as the language is concerned,
-meaning that a Dhall configuration file can hold them and shuffle them around,
-but cannot do anything interesting with them.
+You cannot perform arithmetic on `Double`s.  Only `Natural`
+numbers and `Integer`s support arithmetic and even they only support addition,
+subtraction, and multiplication, because those operations are associative and
```
Comment (sjakobi) [inline docs/discussions/Design-choices.md]: Subtraction isn't associative.

Comment (SiriusStarr) [reply docs/discussions/Design-choices.md]: True!  That was apparently written before `Natural/subtract` was a thing and I didn't catch it, derp.  I'll rewrite per your suggestion, as I think it's better.

**Comment by SiriusStarr** [comment]
@sjakobi Thoughts on the new text?  I mentioned the problem of division by zero for `Natural`/`Integer` and gave a very brief summary of some of the problems with floating point math.  (And we won't talk about the associativity of subtraction...)

File: docs/discussions/Design-choices.md [hunk 2]
```diff
@@ -126,18 +126,16 @@ type).
...
+You cannot perform arithmetic on `Double`s.  `Natural` numbers and `Integer`s
+support addition, subtraction, and multiplication, because those operations are
+well-defined for all possible inputs unlike division (due to the possibility of
+division by zero).
+
+As floating-point arithmetic is imprecise and prone to surprising results (e.g.
+loss of associativity/distributivity, loss of accuracy due to cancellation,
+difficulty testing for equality, etc.), `Double`s are opaque values as far as
+the language is concerned, meaning that a Dhall configuration file can hold them
+and shuffle them around but cannot cannot manipulate them.
```
Comment (sjakobi) [inline docs/discussions/Design-choices.md]:
```suggestion
and shuffle them around but cannot manipulate them.
```

Comment (SiriusStarr) [reply docs/discussions/Design-choices.md]: Squinting in bright sunlight on a laptop screen does not lack of typos make...  >_>

**Comment by Gabriel439** [comment]
@SiriusStarr: This should be good to merge.  As a general rule of thumb you only need to wait 24 hours for changes outside of the standard

**Comment by SiriusStarr** [comment]
@Gabriel439 Yep.  I don't have commit bit for `dhall-lang`, so one of you is going to need to.

**Comment by Gabriel439** [comment]
@SiriusStarr: Whoops!  For some reason I thought you did have the commit bit.  I just gave you write access to the repository so that you can  merge on your own now

**Pull request merged**